src/status_im/contexts/communities/events.cljs

```diff
@@ -188,3 +191,30 @@
 (rf/reg-event-fx :communities/set-addresses-for-permissions
  (fn [{:keys [db]} [addresses]]
    {:db (assoc-in db [:communities/addresses-for-permissions] addresses)}))
+
+(rf/reg-event-fx :communities/share-community-channel-url-with-data
+ (fn [_ [chat-id]]
+   (let [{:keys [community-id channel-id]} (data-store.chats/decode-chat-id chat-id)]
+     {:json-rpc/call [{:method     "wakuext_shareCommunityChannelURLWithData"
```

Comment:
This event could be reformatted a little bit. It has a few non-idiomatic line breaks because zprint is respecting the max line length.

Suggestions:

- Add a line break on the vector passed to `:json-rpc/call`.
- The translation is duplicated a few times, extract that to a local binding.
- We prefer not to use `#()` on longer functions, and prefer it mostly for short, maybe one-liners.

```clojure
(rf/reg-event-fx :communities/share-community-channel-url-with-data
 (fn [_ [chat-id]]
   (let [{:keys [community-id channel-id]} (data-store.chats/decode-chat-id chat-id)
         title                             (i18n/label :t/channel-on-status)]
     {:json-rpc/call
      [{:method     "wakuext_shareCommunityChannelURLWithData"
        :params     [{:CommunityID community-id :ChannelID channel-id}]
        :on-success (fn [url]
                      (share/open
                       (if platform/ios?
                         {:activityItemSources [{:placeholderItem {:type    "text"
                                                                   :content title}
                                                 :item            {:default {:type    "url"
                                                                             :content url}}
                                                 :linkMetadata    {:title title}}]}
                         {:title   title
                          :subject title
                          :message url
                          :url     url})))
        :on-error   (fn [err]
                      (log/error "failed to retrieve community channel url with data"
                                 {:error   err
                                  :chat-id chat-id
                                  :event   "share-community-channel-url-with-data"}))}]})))
```